feat: Add support for custom buckets (#75)
CartoDB/carto-selfhosted-helm: customizations/README.md

> Although we recommend the autoscaling configuration, you could choose the autoscaling feature for some components and the static configuration for the others. Remember that autoscaling override the static configuration, so if one component has both configurations, autoscaling will take precedence.

## Custom Buckets

If you want to keep as much information as possible in your platform you can configure Carto to use your own. Supported storage services are:

+ Google Compute Storage
+ AWS S3
+ Azure Storage

> :warning: You can only set one provider at a time

> These buckets are used as temporary storage when importing data into datawarehouses, for storing maps, thumbnails, and other internal processes data

<!--
TODO: Add the code related to Terraform
-->

### Requirements

> For using you own buckets you need 3 different buckets in your preferred Cloud provider

Reviewer: ```suggestion
You need to create 3 buckets in your preferred Cloud provider
```

+ Import Bucket
+ Client Bucket
+ Thumbnails Bucket

> There's no name constraints

It's mandatory to have credentials for those buckets, our supported credentials methods are

+ GCP: Service Account Key
+ AWS: Access Key ID and Secret Access Key
+ Azure: Storage Access Key

> > :warning: Those credentials should have permissions to interact with the above bucketse

Reviewer: ```suggestion
> :warning: Those credentials should have permissions to interact (read/write) with the above buckets
```

### Google Compute Storage

Add the following lines to your `customizations.yaml`:

```yaml
appConfigValues:
  storageProvider: "gcp"
  importBucket: "carto-import-bucket"
  workspaceImportsBucket: "carto-client-bucket"
  workspaceThumbnailsBucket: "carto-thumbnails-bucket"
  workspaceThumbnailsPublic: false

appSecrets:
  gcpBucketsServiceAccountKey:
    value: |
      {

>       <REDACTED>

Reviewer: ```suggestion
      <REDACTED_JSON_SERVICE_ACCOUNT>
```
should we say it more specific that you need to put the json there?
